OpenRA.Game/FileSystem/MixFile.cs

```diff
@@ -122,6 +122,12 @@ static MemoryStream Decrypt(uint[] h, Blowfish fish)
 
 		static uint[] ReadBlocks(Stream s, long offset, int count)
 		{
+			if (offset > s.Length)
+				throw new ArgumentOutOfRangeException("offset", "Offset {0} greater than stream length {1}.".F(offset, s.Length));
```

Comment:
Looking at this more closely, we may wish to throw an ```ArgumentException``` if offset + bytes to read is greater than the stream length.

We would also want to ensure ```offset``` is non-negative, just like count.